ui/src/app/utils/access-utils.tsx

```diff
@@ -545,34 +546,65 @@ export const isExpiringOrExpired = (
   return !!expiration && getWholeDaysFromNow(expiration) <= lookback;
 };
 
+export const isCompleted = (
+  status: AccessModuleStatus,
+  duccSignedVersion: number
+): boolean =>
+  // special case for DUCC: considered incomplete if the signed version is missing or old
+  status?.moduleName === AccessModule.DATA_USER_CODE_OF_CONDUCT &&
+  !isCurrentDUCCVersion(duccSignedVersion)
+    ? false
+    : !!status?.completionEpochMillis;
+
+export const isBypassed = (status: AccessModuleStatus): boolean =>
+  !!status?.bypassEpochMillis;
+
+export const isCompliant = (
+  status: AccessModuleStatus,
+  duccSignedVersion: number
+): boolean => isCompleted(status, duccSignedVersion) || isBypassed(status);
+
 // is the module "renewal complete" ?
 // meaning (bypassed || (complete and not expiring))
-export const isRenewalCompleteForModule = (status: AccessModuleStatus) => {
-  const isComplete = !!status?.completionEpochMillis;
```

Comment:
removed all the direct accesses like this that I could find

Reply:
Thanks for consolidating these.